File: java/ostalo/Predavanje3_Primjer/app/src/main/java/ada/osc/passwordstealer/PasswordActivity.java
package ada.osc.passwordstealer;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class PasswordActivity extends AppCompatActivity {

    private static final String TAG = PasswordActivity.class.getSimpleName();

    @BindView(R.id.textview_password_passwordDisplay) TextView mPasswordDisplay;
    @BindView(R.id.edittext_password_passwordEntry) EditText mPasswordEntry;
    @BindView(R.id.button_password_passwordSave) Button mPasswordSaver;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_password);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.button_password_passwordSave)
    public void savePassword(){

        String text = mPasswordEntry.getText().toString();
        mPasswordDisplay.setText(text);
        mPasswordEntry.setText(null);

        toastMessage(text);
        logMessage(text);
    }

    private void logMessage(String message) {

        Log.d(TAG, message);
        Log.w(TAG, message);
        Log.e(TAG, message);
    }

    private void toastMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }


}

File: kotlin/ostalo/Predavanje3_Primjer/app/src/main/java/ada/osc/passwordstealer/PasswordActivity.kt
package ada.osc.passwordstealer

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_password.*

class PasswordActivity : AppCompatActivity() {

    //static dio svake klase
    companion object {
        private val TAG = PasswordActivity::class.java.simpleName
    }

    /**
     * Obzirom da koristimo KotlinAndroidExtensions
     * svi view-ovi koji imaju ID u XML-u su automatski bindani za activity/fragment
     * mozemo ih koristiti na nacin da ih pozovemo preko ID-a (npr passwordDisplay).
     *
     * Samim time je bolje view ID-ove nazvati camelCased i po njihovoj funkciji
     * jer su to ipak automatski i nase varijable
     * **/
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_password)

        //ekstenzija za onClickListener
        savePassword.onClick {
            val text = passwordEntry.text.toString()

            passwordDisplay.text = text
            passwordEntry.text = null

            toastMessage(text)
            logMessage(text)
        }
    }

    private fun logMessage(message: String) {
        Log.d(TAG, message)
        Log.w(TAG, message)
        Log.e(TAG, message)
    }

    private fun toastMessage(message: String) {
        //klasican toast
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()

        //toast preko ekstenzija
        toast(message)
    }
}
